app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt

```diff
@@ -33,6 +33,7 @@ class SearchBarHandler(searchBarLayout: ConstraintLayout,
                        checkBoxTitle: String,
                        checkBoxDefaultValue: Boolean,
                        private val _settings: SharedPreferences,
+                       private val _initCallback: (Boolean) -> Unit,
```

Comment:
Note: This change was due to a little bug where `displayContent()` is called twice when starting `Rock-` or `RouteActivity`. However, it somehow belongs to this PR because searching `Alter Weg` on country level requires a lot of computation time (when building all widgets), and with the bug it would even take twice as much.

Reply:
We should think of an issue for resolving this problem. Creating thousands of widgets (which is possible now with route search) is pretty time-consuming. Maybe there are more effective ways like creating only those widgets that are visible in the scroll view?

Reply:
Yeah. I already passed this topic when I implemented the `What's New` dialog. I think the answer is the `ListView` class which handles this for you.